Search buttons and search field are sometimes not visible on Google.com

LumingYin opened the pull request:
#### fef6284d40103b776032d66812367f8f62ec29e7
<pre>
Search buttons and search field are sometimes not visible on Google.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=269055">https://bugs.webkit.org/show_bug.cgi?id=269055</a>
<a href="https://rdar.apple.com/122487319">rdar://122487319</a>

Reviewed by Per Arne Vollan.

* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
Allow looking up `com.apple.MTLCompilerService` without requiring com.apple.webkit.extension.mach
extension only before macOS Sonoma where GPU DOM rendering is not enabled by default.

Canonical link: <a href="https://commits.webkit.org/274634@main">https://commits.webkit.org/274634@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8cec3b194d23ec8853ffecf3a66266fb84be06ab

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39558 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18537 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41912 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42092 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/35458 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21421 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15866 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33037 "Found 1 new test failure: fullscreen/exit-full-screen-video-crash.html (failure)") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40132 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15614 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34243 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13556 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/13531 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43370 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35905 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35525 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/14370 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11836 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15976 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16024 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5196 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/15633 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
```diff
@@ -1381,7 +1381,9 @@
 (allow mach-lookup
     (require-all
+#if !PLATFORM(MAC) || __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000
         (extension "com.apple.webkit.extension.mach")
```
With DOM rendering off, I believe this extension should be present, which should allow the mach-lookup.

LumingYin replied:
I see! Something else may be causing this extension to be intermittently missing when using multiple tabs.

pvollan replied:
Since DOM rendering is off by default on __MAC_OS_X_VERSION_MIN_REQUIRED < 14, I think this patch looks good.

Perhaps it could be changed to `#if PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000`, since Catalyst seems to not be affected by this?

LumingYin replied:
Sounds good, updated PR to use `#if PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000`.

pvollan replied:
I looks like it is not accepted to have a single condition inside `require-all`, so a slight rewrite might be needed.

pvollan replied:
Thanks! To avoid changing for Catalyst, should this be `!PLATFORM(MAC) || __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000` now?

pvollan reviewed (approved):
Thank you! R=me.

Commit:
Search buttons and search field are sometimes not visible on Google.com
https://bugs.webkit.org/show_bug.cgi?id=269055
rdar://122487319

Reviewed by Per Arne Vollan.

* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
Allow looking up `com.apple.MTLCompilerService` without requiring com.apple.webkit.extension.mach
extension only before macOS Sonoma where GPU DOM rendering is not enabled by default.

Canonical link: https://commits.webkit.org/274634@main
